### PR TITLE
[style] updates styling for all auth flows

### PR DESCRIPTION
**Pull request by pragyakallanagoudar**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

This PR updates the styling for all of the auth flows to match Kyrene's newest designs! This involves changing the font to Inter, changing the div for all of the auth screens to have a drop shadow instead of a solid blue border, and other miscellaneous changes like updating spacing, sizing, and adding icons.

i am so cool hehe

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @kyrenetam at the bottom."

Screenshot 2024-04-18 at 9 00 29 PM

Screenshot 2024-04-18 at 9 00 47 PM

Screenshot 2024-04-18 at 9 01 06 PM

Screenshot 2024-04-18 at 9 01 24 PM

Screenshot 2024-04-18 at 9 01 56 PM

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

Run the app, and then navigate to the /login, /signup, and /forgot-password flows. Compare them side-by-side with the Figma to make sure they match!

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

N/A 

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

#76 

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz @kyrenetam

### CODE REVIEW

File: src/app/(auth)/styles.tsx [hunk 1]
```diff
@@ -39,6 +42,19 @@ export const PCentered = styled(P)`
   text-align: center;
 `;
 
+export const AuthHeading = styled.h1`
```
Comment (varortz) [inline src/app/(auth)/styles.tsx]: could we use the `H1` from the global styles? I think its `font-weight` should be `600`, so you can just go ahead and update it.

File: src/app/(auth)/styles.tsx [hunk 2]
```diff
@@ -39,6 +42,19 @@ export const PCentered = styled(P)`
...
+  color: ${COLORS.blueDarker};
+  font-size: 48px;
+  font-style: normal;
+  font-weight: 600;
+`;
+
+export const AuthSubHeading = styled.h2`
+  font-size: 36px;
```
Comment (varortz) [inline src/app/(auth)/styles.tsx]: can we change this to `rem`?

File: src/app/(auth)/forgot-password/page.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ export default function ForgotPassword() {
     <>
       {!emailSentCount && (
         <>
-          <H1 $color={COLORS.blueDarker}>Forgot Password?</H1>
+          <AuthHeading>Forgot Password?</AuthHeading>
```
Comment (varortz) [inline src/app/(auth)/forgot-password/page.tsx]: Not sure if you're done yet with styling since this is a draft PR, but it looks like the text styles aren't being applied here (for the card displayed after the email was sent)? 

Edit: could we also update the button?

**Screenshot from your branch:**
image

**Screenshot from Figma:**
image

File: src/app/(auth)/signup/page.tsx [hunk 4]
```diff
@@ -105,34 +112,33 @@ export default function SignUp() {
...
-              <P $color={COLORS.greyMid}>Email has been resent!</P>
-            )}
-          </HorizontalDiv>
+          </H4Centered>
+          <Button
+            $primaryColor={COLORS.blueMid}
+            $secondaryColor={COLORS.blueDark}
+            onClick={handleResendEmail}
+          >
+            <H4 $color="white">Resend Email</H4>
```
Comment (varortz) [inline src/app/(auth)/signup/page.tsx]: I think we can remove the `H4` tag here

File: src/app/(auth)/signup/page.tsx [hunk 5]
```diff
@@ -105,34 +112,33 @@ export default function SignUp() {
...
+          </Button>
```
Comment (varortz) [inline src/app/(auth)/signup/page.tsx]: Could we update the button to match the designs?

**Screenshot from your branch:**
image

**Screenshot from Figma:**
image

File: src/app/(auth)/styles.tsx [hunk 6]
```diff
@@ -21,15 +21,38 @@ export const FormDiv = styled.div`
...
   display: flex;
   flex-direction: column;
+  ${({ $align }) => $align && `align-items: ${$align};`};
   gap: ${({ $gap }) => $gap || 19.2}px;
 `;
 
 export const H4Centered = styled(H4)`
   color: ${COLORS.greyDark};
   text-align: center;
+  font-size: 16px;
```
Comment (varortz) [inline src/app/(auth)/styles.tsx]: could we update this to `rem` units?

File: src/app/(auth)/styles.tsx [hunk 7]
```diff
@@ -21,15 +21,38 @@ export const FormDiv = styled.div`
...
+  font-style: normal;
+  font-weight: 400;
+`;
+
+export const PCentered = styled(P)`
```
[the remark this answers is not in the document]
Comment (varortz) [reply src/app/(auth)/styles.tsx:39]: actually the `H4Centered` used in the auth flows seems to have a different `font-weight`, so maybe we can keep it in here? should we move `PCentered` to `styles/text.tsx`

File: src/app/(auth)/signup/page.tsx (hunk 4 above)
Comment (varortz) [reply src/app/(auth)/signup/page.tsx]: same for the `forgot-password` page!

File: src/app/(auth)/forgot-password/page.tsx [hunk 8]
```diff
@@ -3,10 +3,10 @@
 import { useState } from 'react';
 import isEmail from 'validator/lib/isEmail';
 import TextInput from '@/components/TextInput/index';
-import { H1, H2, H4, P } from '@/styles/text';
+import { H2, H4, P } from '@/styles/text';
 import supabase from '@/api/supabase/createClient';
 import COLORS from '@/styles/colors';
-import { SpacerDiv, HorizontalDiv } from '@/app/(auth)/styles';
+import { AuthHeading, SpacerDiv, HorizontalDiv } from '@/app/(auth)/styles';
```
Comment (ccatherinetan) [inline src/app/(auth)/forgot-password/page.tsx]: I believe AuthHeading should be `greyDarker`, not `blueDarker` 
nvm it seems like this is a question already on the figma

File: src/app/(auth)/login/page.tsx [hunk 9]
```diff
@@ -65,32 +65,34 @@ export default function Login() {
...
-          <TextInput
-            label="Password"
-            placeholder="Password"
-            errorText={passwordError}
-            type="password"
-            id="password"
-            value={password}
-            setValue={setPassword}
-          />
+          <SpacerDiv $gap={0.813}>
```
Comment (ccatherinetan) [inline src/app/(auth)/login/page.tsx]: I don't think the SpacerDiv here is necessary? (the only child component is the TextInput, so the gap isn't doing anything here either)

File: src/app/(auth)/login/page.tsx [hunk 10]
```diff
@@ -51,7 +51,7 @@ export default function Login() {
   return (
     <>
       <SpacerDiv $gap={0.625}>
-        <H1>Log In</H1>
+        <AuthHeading>Log In</AuthHeading>
         {errorMessage !== '' && <P $color={COLORS.redMid}>{errorMessage}</P>}
       </SpacerDiv>
       <SpacerDiv $gap={0.8125}>
```
Comment (ccatherinetan) [inline src/app/(auth)/login/page.tsx]: I don't think this SpacerDiv is necessary here either, since it is directly nesting another SpacerDiv. 
Additionally, the default gap value of SpacerDiv is 19.2px currently, but I think we want it to be 20px.

File: src/app/(auth)/signup/page.tsx [hunk 11]
```diff
@@ -71,8 +78,8 @@ export default function SignUp() {
     <>
       {!emailSentCount && (
         <>
-          <SpacerDiv $gap={10}>
-            <H1>Sign Up</H1>
+          <SpacerDiv $gap={0.625}>
```
Comment (ccatherinetan) [inline src/app/(auth)/signup/page.tsx]: The value of the gap is in px not rem, if that changes anything.
Also, is the error message supposed to be below instead of above the TextInput? 
Screen Shot 2024-04-18 at 5 32 13 PM

File: src/components/PasswordComplexity.tsx [hunk 12]
```diff
@@ -14,11 +14,11 @@ const PasswordComplexityDiv = styled.div`
 const PasswordRequirementDiv = styled.div<{ met: boolean }>`
   display: flex;
   align-items: center;
-  gap: 8px;
+  gap: 4px;
 `;
 
 const PasswordRequirementText = styled.p<{ met: boolean }>`
```
Comment (ccatherinetan) [inline src/components/PasswordComplexity.tsx]: do we want this to be styled(P) instead?

File: src/styles/fonts.ts [hunk 13]
```diff
@@ -1,3 +1,3 @@
-import { Open_Sans as OpenSans } from 'next/font/google';
+import { Inter } from 'next/font/google';
 
-export const openSans = OpenSans({ subsets: ['latin'] });
+export const sans = Inter({ subsets: ['latin'] });
```
Comment (ccatherinetan) [inline src/styles/fonts.ts]: are we planning to keep the Inter font named as sans?

**Review by ccatherinetan — COMMENTED** [review]
looking good! just a couple questions/catches

File: src/app/(auth)/forgot-password/page.tsx (hunk 8 above)
Comment (pragyakallanagoudar) [reply src/app/(auth)/forgot-password/page.tsx]: so true. ty catherine

File: src/app/(auth)/styles.tsx (hunk 1 above)
Comment (pragyakallanagoudar) [reply src/app/(auth)/styles.tsx]: updated!!

File: src/styles/fonts.ts (hunk 13 above)
Comment (pragyakallanagoudar) [reply src/styles/fonts.ts]: yep! from jinkang:

[jinkang] it's better practice to name the variable according to its style (i.e. sans, sans-serif, display) just so it saves us the many file refactorings later (although we probably won't be the ones benefitting from it)

File: src/app/(auth)/signup/page.tsx (hunk 11 above)
Comment (pragyakallanagoudar) [reply src/app/(auth)/signup/page.tsx]: talked to kyrene and she said to keep the error message at the top!

File: src/app/(auth)/forgot-password/page.tsx (hunk 3 above)
Comment (pragyakallanagoudar) [reply src/app/(auth)/forgot-password/page.tsx]: text weights should be updated now, also i center aligned instead of left aligning to match the Figma!

File: src/app/(auth)/styles.tsx [hunk 14]
```diff
@@ -21,15 +21,42 @@ export const FormDiv = styled.div`
...
+  color: ${COLORS.greyDark};
+  font-size: 1rem;
+  text-align: center;
+`;
+
+export const AuthHeading = styled(H1)`
+  font-size: 48px;
```
Comment (jinkang-0) [inline src/app/(auth)/styles.tsx]: let's use `rem` for font size!

File: src/app/page.module 2.css
Comment (ccatherinetan) [inline src/app/page.module 2.css:1]: can we delete this file?

File: src/components/PasswordComplexity.tsx [hunk 15]
```diff
@@ -11,13 +12,13 @@ const PasswordComplexityDiv = styled.div`
   gap: 4px;
 `;
 
-const PasswordRequirementDiv = styled.div<{ met: boolean }>`
+const PasswordRequirementDiv = styled(P)<{ met: boolean }>`
```
Comment (ccatherinetan) [inline src/components/PasswordComplexity.tsx]: was this supposed to be changed to P, or did you want to keep it as a div? 
Also, if we aren't using `met` for anything, I think we can remove it as a prop from this style.

File: src/app/(auth)/styles.tsx [hunk 16]
```diff
@@ -42,17 +47,16 @@ export const PCentered = styled(P)`
   text-align: center;
 `;
 
-export const AuthHeading = styled.h1`
-  color: ${COLORS.blueDarker};
-  font-size: 48px;
+export const AuthHeading = styled(H1)`
+  font-size: 3rem;
```
Comment (ccatherinetan) [inline src/app/(auth)/styles.tsx]: If this is already a styled `H1`, it will inherit the font size of 3 rem, so we can remove this line. In fact, we do we need this heading at all? I think we can directly use H1.

File: src/app/(auth)/styles.tsx [hunk 17]
```diff
@@ -42,17 +47,16 @@ export const PCentered = styled(P)`
...
   font-style: normal;
-  font-weight: 600;
 `;
 
 export const AuthSubHeading = styled.h2`
```
Comment (ccatherinetan) [inline src/app/(auth)/styles.tsx]: do we want this to be a `styled(H2)`? If so, we can remove the redundant font-size declaration, which would be inherited from H2. I think the only style we'd define is the font-weight. 

I don't think we need text-align either, as long as AuthSubHeading is always wrapped in a center-aligned SpacerDiv

File: src/app/(auth)/signup/page.tsx [hunk 18]
```diff
@@ -76,14 +75,10 @@ export default function SignUp() {
...
-      {!emailSentCount && (
+      {emailSentCount === 0 && (
         <>
-          <SpacerDiv $gap={0.625}>
-            <AuthHeading>Sign Up</AuthHeading>
-            {errorMessage !== '' && (
-              <P $color={COLORS.redMid}>{errorMessage}</P>
-            )}
-          </SpacerDiv>
+          <AuthHeading>Sign Up</AuthHeading>
```
Comment (ccatherinetan) [inline src/app/(auth)/signup/page.tsx]: as mentioned in the other comment, i think we can use H1 here instead of AuthHeading

File: src/app/(auth)/login/page.tsx [hunk 19]
```diff
@@ -50,38 +50,32 @@ export default function Login() {
...
+          label="Password"
+          placeholder="Password"
+          errorText={passwordError}
+          type="password"
+          id="password"
+          value={password}
+          setValue={setPassword}
+        />
+        <P>
+          <LinkColored href="/forgot-password" $color={COLORS.greyMid}>
```
Comment (ccatherinetan) [inline src/app/(auth)/login/page.tsx]: should we add `/forgot-password` and `/signup` (see line 86 below) to CONFIG?

File: src/app/page.module 2.css
Comment (pragyakallanagoudar) [reply src/app/page.module 2.css:1]: yes omg, my GitHub is acting up again ... ty for the catch!

File: src/components/PasswordComplexity.tsx (hunk 15 above)
Comment (pragyakallanagoudar) [reply src/components/PasswordComplexity.tsx]: could not tell u why i randomly decided to do this .... THANKS CATHERINE

File: src/app/(auth)/signup/page.tsx [hunk 20]
```diff
@@ -105,34 +106,29 @@ export default function SignUp() {
...
-            {emailSentCount > 1 && (
-              <P $color={COLORS.greyMid}>Email has been resent!</P>
-            )}
-          </HorizontalDiv>
+          </H4Centered>
+          <BigButton type="button" onClick={handleResendEmail}>
+            Resend Email
+          </BigButton>
+          {emailSentCount > 1 && (
+            <P $color={COLORS.blueMid}>Email has been resent!</P>
```
Comment (jinkang-0) [inline src/app/(auth)/signup/page.tsx]: could we include the user email at the end here?
image

File: src/app/(auth)/signup/page.tsx [hunk 21]
```diff
@@ -66,6 +78,7 @@ export default function SignUp() {
           'https://immigration-justice-project.vercel.app/email-verified',
```
Comment (ccatherinetan) [inline src/app/(auth)/signup/page.tsx]: do we want to use CONFIG here too? i.e. 
`https://immigration-justice-project.vercel.app/{CONFIG.emailverified}`

File: src/lib/icons.tsx [hunk 22]
```diff
@@ -357,6 +357,36 @@ export const IconSvgs = {
       />
     </svg>
   ),
+  blue_check: (
```
Comment (ccatherinetan) [inline src/lib/icons.tsx]: nit: I think the convention was camel case, so it should be `blueCheck`

File: src/app/(auth)/signup/page.tsx [hunk 23]
```diff
@@ -114,30 +127,32 @@ export default function SignUp() {
...
-          )}
-        </HorizontalDiv>
+        </H4Centered>
+        <BigBlueButton type="submit">
+          <H4 $color="white">Resend Email</H4>
+        </BigBlueButton>
+        {emailSentCount > 1 && (
+          <P $color={COLORS.greyDark}>
+            Email has been sent to
+            <span style={{ color: COLORS.blueDark }}>{email}</span>
```
Comment (ccatherinetan) [inline src/app/(auth)/signup/page.tsx]: Screen Shot 2024-05-18 at 11 50 36 PM

Can we add an extra space in before the email?

File: src/app/(auth)/forgot-password/page.tsx [hunk 24]
```diff
@@ -32,39 +32,41 @@ export default function ForgotPassword() {
...
+          <H4Centered $color={COLORS.greyDark}>
+            This link will direct you to the next step. If you didn’t receive an
+            email, please click Resend Email.
+          </H4Centered>
           <BigBlueButton type="submit">
             <H4 $color="white">Resend Email</H4>
           </BigBlueButton>
           {emailSentCount > 1 && (
-            <P $color={COLORS.greyMid}>Email has been resent!</P>
+            <P $color={COLORS.blueMid}>Email has been resent!</P>
```
Comment (ccatherinetan) [inline src/app/(auth)/forgot-password/page.tsx]: to be consistent with the verify email page, are we also planning to display the email address to which the message has been resent?

**Review by ccatherinetan — COMMENTED** [review]
I wasn't able to access the reset password page b/c the forgot password page reroutes to the vercel app, but the rest is looking good! added a few minor nits

File: src/app/(auth)/signup/page.tsx (hunk 21 above)
Comment (varortz) [reply src/app/(auth)/signup/page.tsx]: hm, should we add the whole link to CONFIG?

File: src/app/(auth)/forgot-password/page.tsx (hunk 24 above)
Comment (varortz) [reply src/app/(auth)/forgot-password/page.tsx]: yeah, just updated! thanks for catching this.

File: src/app/(auth)/signup/page.tsx (hunk 21 above)
Comment (ccatherinetan) [reply src/app/(auth)/signup/page.tsx]: Good point; Since this is the only place that we would be routing to the reset-password page (ie we’ll never need a route to ‘/reset-password’), I’m down for adding the entire link to CONFIG!

File: src/app/(auth)/forgot-password/page.tsx [hunk 25]
```diff
@@ -32,39 +32,44 @@ export default function ForgotPassword() {
...
-      <Flex $direction="column" $gap="20px">
-        <H2>A password reset link has been sent to your email.</H2>
-        <H4 $color={COLORS.greyDark}>
-          This link will direct you to the next step. If you didn’t receive an
-          email, please click Resend Email.
-        </H4>
-        <HorizontalDiv>
+      {emailSentCount === 0 ? (
+        <>
+          <H1>Forgot Password?</H1>
```
Comment (ccatherinetan) [inline src/app/(auth)/forgot-password/page.tsx:39]: nit: for the forgot/reset password flow (but not for login/signup), the Figma actually has as font-weight: 500. We could leave it as is, but if we want to change it: 

`<H1 $fontWeight='500'>Forgot Password?</H1>`

**Review by ccatherinetan — APPROVED** [review]
once we decide if we're going to add the reset-password to CONFIG and if we're planning to change the font-weight for the Forgot Password/Reset Password flow, it looks good to merge !

File: src/app/(auth)/forgot-password/page.tsx (hunk 25 above)
Comment (varortz) [reply src/app/(auth)/forgot-password/page.tsx:39]: Probably best if we stay consistent (`600px`), especially since it's a pretty similar design.

File: src/app/(auth)/signup/page.tsx (hunk 21 above)
Comment (varortz) [reply src/app/(auth)/signup/page.tsx]: added!

**Review by varortz — APPROVED** [review]
THANK YOU PRAGYA FOR FINISHING OFF OUR STYLING & THANK YOU JINKANG FOR REBASING!!
